### PR TITLE
firebase_options.dart, gradle, dependabot.yml, markdown

### PR DESCRIPTION
**Pull request by StructPoltergeist**
# Workflow

## Analysis

```shell
flutter analyze lib

Analyzing lib...                                                

   info • Use 'const' with the constructor to improve performance • lib/src/config/firebase_options.dart:49:32 • prefer_const_constructors
   info • Use 'const' with the constructor to improve performance • lib/src/config/firebase_options.dart:58:36 • prefer_const_constructors
   info • Use 'const' with the constructor to improve performance • lib/src/config/firebase_options.dart:66:32 • prefer_const_constructors

3 issues found. (ran in 8.7s)
Error: Process completed with exit code 1.
```

- firebase_options.dart - (prefer_const_constructors)

- https://github.com/HelpAPaw/Flutter/actions/workflows/flutter.yml

## Build 

- build.gradle - (Gradle Configuration)

- gradle.properties -  (Gradle Properties Configuration) 

- gradle-wrapper.properties -  (Gradle Wrapper Configuration)

- settings.gradle -  (Gradle Settings Configuration) 

## Dependabot

- dependabot.yml - (Automated dependency updates built into GitHub)
 
- https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file
 
- https://github.com/dependabot

### CODE REVIEW

File: android/gradle.properties [hunk 1]
```diff
@@ -1,6 +1,5 @@
 org.gradle.jvmargs=-Xmx1536M
 android.useAndroidX=true
 android.enableJetifier=true
-android.defaults.buildfeatures.buildconfig=true
```
Comment (BurningAXE) [inline android/gradle.properties:4]: Why was this deleted?

**Review by BurningAXE — COMMENTED** [review]
Thanks!
Can you please squash all commits into a single one, so history is cleaner after merge.

File: android/gradle.properties (hunk 1 above)
Comment (StructPoltergeist) [reply android/gradle.properties:4]:
```groovy
buildscript {
    dependencies {
        classpath 'com.android.tools.build:gradle:8.2.2'
    }
}
```

## Deprecated

```groovy
android.defaults.buildfeatures.buildconfig=true
```

## Implemented

```groovy
android {
    buildFeatures {
        buildConfig = true
    }
}
```

**Comment by StructPoltergeist** [comment]
- com.android.tools.build >= 8.0.0

- com.android.application >= 8.0.0

- com.android.library >= 8.0.0

**Review by BurningAXE — DISMISSED** [review]

**Comment by BurningAXE** [comment]
Was waiting for you to squash the commits...
